l10n_nl_cbs_export/models/cbs_export_file.py

```diff
@@ -0,0 +1,146 @@
+        ('06', 'June'),
+        ('07', 'July'),
+        ('08', 'August'),
+        ('09', 'September'),
+        (10, 'October'),
+        (11, 'November'),
+        (12, 'December')
+    ], default='01', string='Month')
+    year = fields.Char('Year')
+    account_invoice_ids = fields.One2many('account.invoice', 'cbs_export_id')
+
+    @api.model
+    def create(self, vals):
+        vals['name'] = self.env['ir.sequence'].next_by_code('cbs.export.file')
+        return super(cbs_export_file, self).create(vals)
+
+    @api.one
+    def get_data(self):
+        self.set_invoice()
+        self.export_file()
+
+    @api.constrains('year')
+    def check_year(self):
+        if self.year:
+            if not self.year.isdigit():
+                raise ValidationError(_("Please insert a valid Year"))
+            else:
+                is_valid_year = '%d' % (int(self.year))
+                try:
+                    time.strptime(is_valid_year, '%Y')
+                except ValueError:
+                    raise ValidationError(_("Please insert a valid Year"))
+
+    @api.one
+    def set_invoice(self):
+        days = calendar.monthrange(int(self.year), int(self.month))
+        account_ids = self.env['account.invoice'].search([
+            ('type', '=', 'out_invoice'),
+            ('state', 'in', ['open', 'paid']),
+            ('partner_id.country_id.intrastat', '=', True),
+            ('partner_id.country_id.code', '!=', 'NL'),
+            ('date_invoice', '>=', datetime.strptime(
+                '%s-%s-%s' % (1, int(self.month), int(self.year)), '%d-%m-%Y')),
+            ('date_invoice', '<=', datetime.strptime(
+                '%s-%s-%s' % (days[1], int(self.month), int(self.year)), '%d-%m-%Y'))])
+        if not account_ids:
+            raise ValidationError(
+                _("There are no invoice lines for CBS Export during month %s in year %s") % (calendar.month_name[int(self.month)], self.year)
+            )
+        else:
+            account_ids.search([('cbs_export_id', '=', self.id)]).write(
+                {'cbs_export_id': False}
+            )
+            account_ids.write({'cbs_export_id': self.id})
+
+    @api.model
+    def cron_get_cbs_export_file(self):
+        last_month = date.today().replace(day=1) - timedelta(days=1)
+        sr_ids = self.search([('month', '=', last_month.strftime("%m")), ('year', '=', last_month.strftime("%Y"))])
+        if not sr_ids:
+            sr_ids = self.create({'month': last_month.strftime("%m"), 'year': last_month.strftime("%Y")})
+            sr_ids.message_post(
+                body=_("CBS Export is created for month %s in year %s") % (calendar.month_name[int(sr_ids.month)], sr_ids.year),
+                subtype='mt_comment'
+            )
+        sr_ids.set_invoice()
+        sr_ids.export_file()
+
+    @api.one
+    def export_file(self):
+        line_number = 1
+        company_id = self.env.user.company_id
+        cbs_export_data = str('9801')+str(
```

Comment:
it gives other developers a modular way to hook into the generation process and modify it. So it's needed to make your code better.